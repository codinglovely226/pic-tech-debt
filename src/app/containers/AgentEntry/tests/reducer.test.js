/* eslint-disable */

import agentsReducer from '../reducer';

import {
  addAgent,
  deleteAgent,
} from '../actions/case';

describe('agentsReducer', () => {
  it('returns the initial state', () => {
    expect(agentsReducer(undefined, {})).toMatchSnapshot();
  });
  it('add agent', () => {
    expect(agentsReducer([{'id': 1}], addAgent(2, 'Test Agent', '1', 'mg', false, '2018-04-14 14:25:00', '1', 'hour', false, 'Test Exposure Type', 'Ingested', '123456'))).toMatchSnapshot();
  });
  it('delete agent', () => {
    expect(agentsReducer([{'id': 1}], deleteAgent(1, 1, 'Test Agent'))).toMatchSnapshot();
  });
});
