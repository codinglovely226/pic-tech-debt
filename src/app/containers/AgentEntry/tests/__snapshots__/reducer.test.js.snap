exports[`agentsReducer add agent 1`] = `
Array [
  Object {
    "id": 1,
  },
  Object {
    "category": "",
    "centreAgents": Array [],
    "dosage": Object {
      "quantity": "1",
      "units": "mg",
      "unknown": false,
    },
    "eupcs": "",
    "exposure": Object {
      "duration": "1",
      "durationUnits": "hour",
      "durationUnknown": false,
      "route": "Ingested",
      "timeSince": "",
      "timeSinceUnits": "",
      "timestamp": "2018-04-14 14:25:00",
      "type": "Test Exposure Type",
    },
    "id": "123456",
    "intendedUseGiven": "",
    "isPatientsOwnMedication": false,
    "name": "Test Agent",
    "type": Array [],
  },
]
`;

exports[`agentsReducer delete agent 1`] = `Array []`;

exports[`agentsReducer returns the initial state 1`] = `Array []`;
